server/users.js

```diff
@@ -430,89 +430,84 @@ exports.updateProfile = function (userId, request, callback) {
-             */
-            if (userind < 6) {
-                for (var i = 0; i < 7; ++i) {
-                    lb.push(studentlist[i]);
+                var student = {
+                    displayName:`${currentStudent.fname} ${currentStudent.lname[0]}.`,
+                    points:currentStudent.points,
+                    userRank: rank
                 }
-                lb.push(dots);
-            } else {
-                /* The top 3 students are always displayed. */
-                lb.push(studentlist[0]);
-                lb.push(studentlist[1]);
-                lb.push(studentlist[2]);
-
-                /*
-                 * If the user is in the bottom four,
-                 * display the whole bottom four.
-                 */
-                if (userind >= studentlist.length - 4) {
-                    lb.push(dots);
-                    for (var i = studentlist.length - 4; i < studentlist.length; ++i) {
-                        lb.push(studentlist[i]);
-                    }
-                } else {
-                    /*
-                     * Otherwise, display the user with the
-                     * students directly above and below them.
-                     */
-                    lb.push(dots);
-                    lb.push(studentlist[userind - 1]);
-                    lb.push(studentlist[userind]);
-                    lb.push(studentlist[userind + 1]);
-                    lb.push(dots);
+
+                // Push only students with the top 3 number of poitns to mini leaderboard
+                if (i < 3) {
+                    leaderboardList.push(student);
+                }
+            }
+        }
+
+        else {
+            for (var i = 0; i < studentlist.length; ++i) {
+                let currentStudent = studentlist[i];
+
+                var student = {
+                    displayName:`${currentStudent.fname} ${currentStudent.lname[0]}.`,
+                    points:currentStudent.points,
+                    accuracy:(currentStudent.totalAttemptsCount === 0)
+                        ? 0
+                        : ((currentStudent.correctAttemptsCount / currentStudent.totalAttemptsCount) * 100).toFixed(2),
+                    attempt:(currentStudent.totalAttemptsCount === 0)
+                        ? 0
+                        : (currentStudent.points / currentStudent.totalAttemptsCount).toFixed(2),
+                    overall:(currentStudent.totalAttemptsCount === 0)
+                        ? 0
+                        : (currentStudent.points * 
+                        ((currentStudent.correctAttemptsCount/currentStudent.totalAttemptsCount) + 
+                        (currentStudent.points/currentStudent.totalAttemptsCount))).toFixed(2)
                 }
+
+                leaderboardList.push(student);
             }
-            callback(lb);
-        } else {
-            callback(studentlist);
         }
+        callback(leaderboardList, displayRank);
```

Comment:
just to be consistent please return an err, you can send null back

Reply:
Fixed